rope/base/prefs.py

```diff
@@ -12,6 +12,14 @@
 from rope.base.resources import Folder
 
 
+@dataclass
+class AutoimportPrefs:
+    underlined: bool = field(
+        default=False, description="Cache underlined (private) modules")
+    memory: bool = field(default=None, description="Cache in memory instead of disk")
+    parallel: bool = field(default=True, description="Use multiple processes to parse")
```

Comment:
Thinking about this further on the config.

First, I don't think `parallel` should be user-configurable. I'd have preferred to not make parallelization configurable at all, but if there's use case single-threaded mode, IMO it's generally the editor/IDE integrators that should choose the best value for their integration, not the user.

Second, I'm not quite sure whether `memory` should be global or autoimport-specific configuration. While this setting currently only affects autoimport, we may want to expand this to other caches. Users that don't want rope to create any cache files likely would want to just turn off all cached files and users that wanted caches mostly wouldn't care about the different cache files that rope generates. 

It's only a small amount of very advanced users or users that are having very specific problems that would've cared about toggling these settings per internal rope component. Unless there's a strong use case for per-component fine tuning here, I would prefer to keep things that can be configured to a minimum to keep maintenance simpler.